### PR TITLE
DAOS-12407 cart: Update libfabric to 1.17.0

### PR DESCRIPTION
**Pull request by frostedcmos**
- Update to v1.17.0

PR-repos: libfabric@PR-66

Required-githooks: true

Signed-off-by: Alexander A Oganezov <alexander.a.oganezov@intel.com>

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by brianjmurrell** [comment]
FWIW: having to create this kind of PR is exactly what [`packageBuildingPipelineDAOSTest()`](https://github.com/daos-stack/pipeline-lib/blob/master/vars/packageBuildingPipelineDAOSTest.groovy) is intended to avoid.  If you want to simply run the `pr` set (as you are doing here) using `packageBuildingPipelineDAOSTest()` in the libfabric repo you could just set `packageBuildingPipelineDAOSTest()`'s `test-tag` argument to `pr`.

**Comment by frostedcmos** [comment]
> FWIW: having to create this kind of PR is exactly what [`packageBuildingPipelineDAOSTest()`](https://github.com/daos-stack/pipeline-lib/blob/master/vars/packageBuildingPipelineDAOSTest.groovy) is intended to avoid. If you want to simply run the `pr` set (as you are doing here) using `packageBuildingPipelineDAOSTest()` in the libfabric repo you could just set `packageBuildingPipelineDAOSTest()`'s `test-tag` argument to `pr`.

Ah, i was not aware of 'pr' test-tag, and was not sure what all tags to include to run a standard daos build. will do next update:)

File: utils/docker/Dockerfile.el.8 [hunk 1]
```diff
@@ -137,9 +137,9 @@ ARG JOBS=$DEPS_JOBS
 ARG DAOS_BUILD_TYPE=$DAOS_TARGET_TYPE
 ARG DAOS_BUILD=$DAOS_DEPS_BUILD
 
-# Build DAOS
+# Build DAOS with STACK_MMAP=1
 RUN [ "$DAOS_BUILD" != "yes" ] || {                                        \
-        scons --jobs $JOBS install PREFIX=/opt/daos COMPILER=$COMPILER     \
+        scons --jobs $JOBS install STACK_MMAP=1 PREFIX=/opt/daos COMPILER=$COMPILER     \
```
Comment (jolivier23) [inline utils/docker/Dockerfile.el.8:142]: why not just change the default?

Comment (daltonbohning) [reply utils/docker/Dockerfile.el.8:142]: Until I complete https://daosio.atlassian.net/browse/DAOS-11547, we're concerned about the performance impact of `STACK_MMAP=1`

**Pull request closed without merging**